base/acme/src/main/java/org/dogtagpki/acme/server/ACMEDirectoryService.java

```diff
@@ -36,21 +37,30 @@ public Response getDirectory() throws Exception {
 
         logger.info("Creating directory");
 
+        ACMEEngine engine = ACMEEngine.getInstance();
+        URI baseURL = engine.getBaseURL();
+
+        UriBuilder uriBuilder;
+        if (baseURL != null) {
```

Comment:
Actually, since the `baseURL` is supposed to point to an actual location instead of just an identifier, I changed the type from `URI` into `URL`. See `ACMEEngineConfig.fromProperties()`. The `URL` constructor there will reject empty/invalid base URL in `engine.conf`.